staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -539,21 +542,25 @@ func (p *processorListener) pop() {
 	}
 }
 
-func (p *processorListener) run() {
-	defer utilruntime.HandleCrash()
-
-	for next := range p.nextCh {
-		switch notification := next.(type) {
-		case updateNotification:
-			p.handler.OnUpdate(notification.oldObj, notification.newObj)
-		case addNotification:
-			p.handler.OnAdd(notification.newObj)
-		case deleteNotification:
-			p.handler.OnDelete(notification.oldObj)
-		default:
-			utilruntime.HandleError(fmt.Errorf("unrecognized notification: %#v", next))
+func (p *processorListener) run(stopCh <-chan struct{}) {
+	// this call blocks until the channel is closed.  When a panic happens during the notification
+	// we will catch it, **the offending item will be skipped!**, and after a short delay (one second)
+	// the next notification will be attempted.  This is usually better than the alternative of never
```

Comment:
Hmmm... if there's a more permanent problem with delivering, this will keep failing every second, why not using some backoff here as well?